docs/src/main/paradox/atleastonce.md

```diff
@@ -0,0 +1,78 @@
+
+This is a significant challenge. Below we suggest a few strategies to deal with some special cases of this general problem.
+ 
+### Publishing to Message-Dependent Topics
+
+Since `ProducerRecord` contains the destination topic, it is possible to use a single producer flow to write to any number of topics. This preserves the ordering of messages coming from the committable source. Since the destination topics likely admit different types of messages, it will be necessary to serialize the messages to the appropriate input type for the common producer flow, which could be a byte array or a string.
+
+Each committable message may safely lead to the production of more than one message, as long as the `CommittableOffset` is associated to the last message, as described earlier.
+
+However if a committable message leads to the production of no messages, then we have a problem: the producer flow is not currently able to pass through a committable offset without producing a message.
+
+### Excluding Messages
+
+Failure to deserialize a message is a particular case of conditional message processing. It is also likely that we would have no message to produce to Kafka when we encounter messages that fail to deserialize. As described above, the producer flow will not less us pass through the corresponding committable offset without producing a message. 
+
+Why can't we commit the offsets of bad messages as soon as we encounter them, instead of try passing them on, downstream? Because the previous offsets, for messages that have deserialized successfully, may not have been committed yet. That's possible if the downstream flow includes a buffer, an aynchronous boundary or performs batching. It is then likely that some previous messages would concurrently be making their way downstream to a final committing stage.
+
+Note that here we assume that we take the full control over the handling of messages that fail to deserialize. To do this, we should not ask for the deserialization to be performed by the commitable source. We can instead create a `ConsumerSettings` parametrized by byte arrays. A subsequent `collect` can deserialize and skip bad messages. Alternatively a `map` stage can be used should we wish to propagate downstream some information about the bad messages, such as their committable offsets.
+
+If bad messages are rare, it might be acceptable to never commit their offsets directly and instead rely on the commit of the next deserializable message to eventually advance the partition beyond the bad messages. This preserves at-least-once semantics, but can lead to more frequent duplicated handling of bad messages on restarts. However that handling might may not have very important effects: it might simply be logging the message.
```

Comment:
Well said. We've reached a conclusion while dealing with this in our applications that the `Producer.flow` interface is too simplistic, when instead having a producer function that you can run with `mapAsync` (and optionally skip producing if encountering a `None`) allows for the control needed in this case.